### ISSUE TITLE
Crash when looking for starbound folder

### ISSUE THREAD

**Issue by blacksteve13** [wizzomafizzo/starcheat#121]
Everytime i try and the starbound folder i get this error 

Traceback (most recent call last):
  File "C:\starcheat\build\starcheat.py", line 74, in <module>
    main()
  File "C:\starcheat\build\starcheat.py", line 71, in main
    gui.mainwindow.MainWindow()
  File "C:\starcheat\build\gui\mainwindow.py", line 61, in __init__
    if not new_setup_dialog(self.window):
  File "C:\starcheat\build\gui\utils.py", line 184, in new_setup_dialog
    if not build_assets_db(parent):
  File "C:\starcheat\build\gui\utils.py", line 57, in build_assets_db
    asset_files = assets_db.find_assets()
  File "C:\starcheat\build\assets.py", line 214, in find_assets
    vanilla_assets = self.scan_asset_folder(vanilla_path)
  File "C:\starcheat\build\assets.py", line 248, in scan_asset_folder
    files = os.listdir(folder)
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'H:\\SteamLibrary\\SteamApps\\common\\Starbound\\giraffe_storage\\assets'

**Comment by chrmoritz** [comment]
Try to use a [version with Upbeat Giraffe support](https://github.com/wizzomafizzo/starcheat/releases/tag/0.18).

**Comment by blacksteve13** [comment]
I am using that version and that still doesnt work

**Comment by blacksteve13** [comment]
everytime i start star cheat it says "Starbound assets have been changed  rebuld the starcheat assets index "



If i press yes i get this error 

Traceback (most recent call last):
  File "C:\starcheat\build\starcheat.py", line 74, in <module>
    main()
  File "C:\starcheat\build\starcheat.py", line 71, in main
    gui.mainwindow.MainWindow()
  File "C:\starcheat\build\gui\mainwindow.py", line 67, in __init__
    if not check_index_valid(self.window):
  File "C:\starcheat\build\gui\utils.py", line 95, in check_index_valid
    return build_assets_db(parent)
  File "C:\starcheat\build\gui\utils.py", line 57, in build_assets_db
    asset_files = assets_db.find_assets()
  File "C:\starcheat\build\assets.py", line 214, in find_assets
    vanilla_assets = self.scan_asset_folder(vanilla_path)
  File "C:\starcheat\build\assets.py", line 241, in scan_asset_folder
    db = starbound.open_file(pak_path)
  File "C:\starcheat\build\starbound\helpers.py", line 213, in open
    return read_stream(builtins.open(path, 'rb'), extension)
  File "C:\starcheat\build\starbound\helpers.py", line 220, in read_stream
    file.initialize()
  File "C:\starcheat\build\starbound\btreedb4.py", line 119, in initialize
    super(FileBTreeDB4, self).initialize()
  File "C:\starcheat\build\starbound\sbbf02.py", line 89, in initialize
    assert stream.read(6) == b'SBBF02', 'Invalid file format'
AssertionError: Invalid file format





if i press no i get this error 


Traceback (most recent call last):
  File "C:\starcheat\build\starcheat.py", line 74, in <module>
    main()
  File "C:\starcheat\build\starcheat.py", line 71, in main
    gui.mainwindow.MainWindow()
  File "C:\starcheat\build\gui\mainwindow.py", line 98, in __init__
    for species in self.assets.species().get_species_list():
  File "C:\starcheat\build\assets.py", line 350, in species
    return Species(self)
  File "C:\starcheat\build\assets.py", line 931, in __init__
    self.assets.vanilla_assets)
  File "C:\starcheat\build\assets.py", line 306, in read
    db = starbound.open_file(path)
  File "C:\starcheat\build\starbound\helpers.py", line 213, in open
    return read_stream(builtins.open(path, 'rb'), extension)
  File "C:\starcheat\build\starbound\helpers.py", line 220, in read_stream
    file.initialize()
  File "C:\starcheat\build\starbound\btreedb4.py", line 119, in initialize
    super(FileBTreeDB4, self).initialize()
  File "C:\starcheat\build\starbound\sbbf02.py", line 89, in initialize
    assert stream.read(6) == b'SBBF02', 'Invalid file format'
AssertionError: Invalid file format

**Comment by chrmoritz** [comment]
Then fix your assets location setting or [reset your config](https://github.com/wizzomafizzo/starcheat/wiki/help#reset-all-settings)

**Comment by blacksteve13** [comment]
I've been using the latest version the whole time. I even re-installed python. I downloaded the "py-starbound master" (i dont know if that is a problem)  I did the reset your config but I dont exactly know how to fix the assets location setting 

I did the reset your config thing and now i keep getting this
Traceback (most recent call last):
  File "C:\starcheat\build\starcheat.py", line 74, in <module>
    main()
  File "C:\starcheat\build\starcheat.py", line 71, in main
    gui.mainwindow.MainWindow()
  File "C:\starcheat\build\gui\mainwindow.py", line 61, in __init__
    if not new_setup_dialog(self.window):
  File "C:\starcheat\build\gui\utils.py", line 184, in new_setup_dialog
    if not build_assets_db(parent):
  File "C:\starcheat\build\gui\utils.py", line 57, in build_assets_db
    asset_files = assets_db.find_assets()
  File "C:\starcheat\build\assets.py", line 214, in find_assets
    vanilla_assets = self.scan_asset_folder(vanilla_path)
  File "C:\starcheat\build\assets.py", line 248, in scan_asset_folder
    files = os.listdir(folder)
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'H:\\SteamLibrary\\SteamApps\\common\\Starbound\\giraffe_storage\\assets'

**Comment by chrmoritz** [comment]
Nope, you are using a old version: The `SBBF02` format was used in the old unstable/nightly builds of starbound. Upbeat Giraffe uses `SBBF03` and support for it was added in 01f5d01 by a submodule change/update. So you have either to update the repo and submodule or use the binaries I've linked in my first post.

**Comment by blacksteve13** [comment]
So just download starcheat-0.18 and use the updated  py-starbound and i should be good

**Comment by blacksteve13** [comment]
i just used your binaries and now im getting this error 



aceback (most recent call last):
  File "C:\Python34\lib\site-packages\cx_Freeze\initscripts\Console.py", line 27, in <module>
    exec(code, m.__dict__)
  File "build\starcheat.py", line 74, in <module>
  File "build\starcheat.py", line 71, in main
  File "build\gui\mainwindow.py", line 61, in __init__
  File "build\gui\utils.py", line 184, in new_setup_dialog
  File "build\gui\utils.py", line 57, in build_assets_db
  File "build\assets.py", line 214, in find_assets
  File "build\assets.py", line 248, in scan_asset_folder
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'H:\\SteamLibrary\\SteamApps\\common\\Starbound\\giraffe_storage\\assets'

**Comment by chrmoritz** [comment]
[reset your config](https://github.com/wizzomafizzo/starcheat/wiki/help#reset-all-settings).

**Comment by blacksteve13** [comment]
Stil getting the same error 


Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\cx_Freeze\initscripts\Console.py", line 27, in <module>
    exec(code, m.__dict__)
  File "build\starcheat.py", line 74, in <module>
  File "build\starcheat.py", line 71, in main
  File "build\gui\mainwindow.py", line 61, in __init__
  File "build\gui\utils.py", line 184, in new_setup_dialog
  File "build\gui\utils.py", line 57, in build_assets_db
  File "build\assets.py", line 214, in find_assets
  File "build\assets.py", line 248, in scan_asset_folder
FileNotFoundError: [WinError 3] The system cannot find the path specified: 'H:\\SteamLibrary\\SteamApps\\common\\Starbound\\giraffe_storage\\assets'

**Comment by chrmoritz** [comment]
Whats the content of the `starcheat.ini`?

**Comment by blacksteve13** [comment]
[starcheat]
pak_hash = none
config_version = 10
backup_folder = C:\Users\BlackSteve\AppData\Roaming\starcheat\backups
assets_db = C:\Users\BlackSteve\AppData\Roaming\starcheat\assets.db
assets_folder = H:\SteamLibrary\SteamApps\common\Starbound\giraffe_storage\assets
player_folder = H:\SteamLibrary\SteamApps\common\Starbound\giraffe_storage\giraffe_storage\player
starbound_folder = H:\SteamLibrary\SteamApps\common\Starbound\giraffe_storage

**Comment by chrmoritz** [comment]
Change it to
```
[starcheat]
pak_hash = none
config_version = 10
backup_folder = C:\Users\BlackSteve\AppData\Roaming\starcheat\backups
assets_db = C:\Users\BlackSteve\AppData\Roaming\starcheat\assets.db
assets_folder = H:\SteamLibrary\SteamApps\common\Starbound\assets
player_folder = H:\SteamLibrary\SteamApps\common\Starbound\giraffe_storage\player
starbound_folder = H:\SteamLibrary\SteamApps\common\Starbound
```

**Comment by blacksteve13** [comment]
Okay that worked. IS there something i did wrong

**Comment by chrmoritz** [comment]
How did you set the Starbound folder for starcheat? Looking at the auto detect code it seems very unlikely that if have detected `H:\SteamLibrary\SteamApps\common\Starbound\giraffe_storage` instead of `H:\SteamLibrary\SteamApps\common\Starbound` by itself.

**Comment by blacksteve13** [comment]
Everytime i select H:\SteamLibrary\SteamApps\common\Starbound it says thats not my folder because it doesnt have the starbound.config file

**Comment by chrmoritz** [comment]
This should be fixed now in 0.18.1 (Upbeat Giraffe has changed the location of the starbound.config file). And 0.18.1 should now auto detect starbound correctly if it is not in the main steam library as in your case.
